pymeasure/instruments/instrument.py

```diff
@@ -149,134 +257,200 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
+                command_process=lambda c: c,
+                check_set_errors=False,
+                check_get_errors=False,
+                dynamic=False,
                 **kwargs):
         """Returns a property for the class based on the supplied
         commands. This property may be set and read from the
-        instrument.
+        instrument. See also :meth:`measurement` and :meth:`setting`.
 
-        :param get_command: A string command that asks for the value
-        :param set_command: A string command that writes the value
+        :param get_command: A string command that asks for the value, set to `None`
+            if get is not supported (see also :meth:`setting`).
+        :param set_command: A string command that writes the value, set to `None`
+            if set is not supported (see also :meth:`measurement`).
         :param docs: A docstring that will be included in the documentation
         :param validator: A function that takes both a value and a group of valid values
-                          and returns a valid value, while it otherwise raises an exception
+            and returns a valid value, while it otherwise raises an exception
         :param values: A list, tuple, range, or dictionary of valid values, that can be used
-                       as to map values if :code:`map_values` is True.
+            as to map values if :code:`map_values` is True.
         :param map_values: A boolean flag that determines if the values should be
-                          interpreted as a map
+            interpreted as a map
         :param get_process: A function that take a value and allows processing
-                            before value mapping, returning the processed value
+            before value mapping, returning the processed value
         :param set_process: A function that takes a value and allows processing
-                            before value mapping, returning the processed value
+            before value mapping, returning the processed value
+        :param command_process: A function that take a command and allows processing
+            before executing the command, for getting
         :param check_set_errors: Toggles checking errors after setting
         :param check_get_errors: Toggles checking errors after getting
+        :param dynamic: Specify whether the property parameters are meant to be changed in
+            instances or subclasses.
+
+        Example of usage of dynamic parameter is as follow:
+
+        .. code-block:: python
+
+            class GenericInstrument(Instrument):
+                center_frequency = Instrument.control(
+                    ":SENS:FREQ:CENT?;", ":SENS:FREQ:CENT %e Hz;",
+                    " A floating point property that represents the frequency ... ",
+                    validator=strict_range,
+                    # Redefine this in subclasses to reflect actual instrument value:
+                    values=(1, 26.5e9),
```

Comment:
maybe for the example here it would be simpler/clearer to have "easier" values here, like you chose in the tests, e.g. `(1, 1000)`, etc.

Reply:
Indeed, good suggestion